src/NuGet.Core/NuGet.ProjectModel/PackageSpecOperations.cs

```diff
@@ -26,7 +26,7 @@ public static void AddOrUpdateDependency(PackageSpec spec, PackageDependency dep
 
             if (!existing.Any())
             {
-                AddDependency(spec.Dependencies, dependency.Id, range);
+                AddDependency(spec.Dependencies, dependency.Id, range, spec.RestoreMetadata?.CentralPackageVersionsEnabled ?? false);
```

Comment:
This modification is already in review based on @jeffkl PR (https://github.com/NuGet/NuGet.Client/pull/4642) so I will remove these changes once his PR is merged into dev.